artichoke-backend/src/extn/stdlib/strscan/strscan.rb

```diff
@@ -49,40 +49,45 @@ def [](group)
   end
 
   def beginning_of_line?
-    return true if @charpos.zero?
+    return true if @pos.zero?
 
-    @string[@charpos - 1] == "\n"
+    @string.byteslice(@pos - 1) == "\n"
   end
   alias bol? beginning_of_line?
 
   def captures
     @last_match&.captures
   end
 
+  def charpos
+    @string.byteslice(0, @pos).length
+  end
+
   def charpos=(pointer)
```

Comment:
this is not a real method in the `StringScanner` class:

- https://ruby-doc.org/stdlib-2.6.3/libdoc/strscan/rdoc/StringScanner.html
- https://ruby-doc.org/stdlib-3.1.0/libdoc/strscan/rdoc/StringScanner.html